frontend/src/views/FuelCodes/AddFuelCode/AddEditFuelCode.jsx

```diff
@@ -0,0 +1,373 @@
+import { useCallback, useEffect, useRef, useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import { Stack, Typography } from '@mui/material'
+import Grid2 from '@mui/material/Unstable_Grid2/Grid2'
+import { v4 as uuid } from 'uuid'
+import { BCGridEditor } from '@/components/BCDataGrid/BCGridEditor'
+import Loading from '@/components/Loading'
+import { roles } from '@/constants/roles'
+import { ROUTES } from '@/constants/routes'
+import {
+  useApproveFuelCode,
+  useCreateFuelCode,
+  useDeleteFuelCode,
+  useFuelCodeOptions,
+  useGetFuelCode,
+  useUpdateFuelCode
+} from '@/hooks/useFuelCode'
+import withRole from '@/utils/withRole'
+import { defaultColDef, fuelCodeColDefs } from './_schema'
+import { useParams } from 'react-router-dom'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { faFloppyDisk } from '@fortawesome/free-solid-svg-icons'
+import BCButton from '@/components/BCButton'
+import BCModal from '@/components/BCModal'
+import BCTypography from '@/components/BCTypography'
+import { FUEL_CODE_STATUSES } from '@/constants/statuses'
+import { useCurrentUser } from '@/hooks/useCurrentUser'
+
+const AddEditFuelCodeBase = () => {
+  const { fuelCodeID } = useParams()
+  const gridRef = useRef(null)
+  const alertRef = useRef()
+  const { t } = useTranslation(['common', 'fuelCode'])
+
+  const [rowData, setRowData] = useState([])
+  const [errors, setErrors] = useState({})
+  const [columnDefs, setColumnDefs] = useState([])
+  const [isGridReady, setGridReady] = useState(false)
+  const [modalData, setModalData] = useState(null)
+
+  const { hasRoles } = useCurrentUser()
+  const { data: optionsData, isLoading, isFetched } = useFuelCodeOptions()
+  const { mutateAsync: updateFuelCode } = useUpdateFuelCode(fuelCodeID)
```

Comment:
Does it make sense to have any isLoading checks on these crud mutations?

Reply:
Looked into this, BCModal is setting a loading boolean for Approve Code that prevents double actions from firing. I don't think we need one for the non-modal use cases are they use the standard ag-grid approach.